Updates on Cache Busting (#296)
commerce-docs/microsite-commerce-storefront: src/content/docs/setup/seo/indexing.mdx

  This is required because Google caches responses without considering headers. Not adding the store code to the URL might lead to Google indexing the wrong data. You can ensure this by adding a cache-busting query parameter.
  :::

### Cache-busting Query Parameter

> The [Commerce boilerplate template](https://github.com/hlxsites/aem-boilerplate-commerce/tree/main) implements a dynamic cache-busting mechanism for the Catalog Service endpoint. This ensures that when configuration headers change, the browser will fetch fresh data instead of using cached responses.

Reviewer: ```suggestion
The [Commerce boilerplate template](https://github.com/hlxsites/aem-boilerplate-commerce/tree/main) includes a built-in way to prevent stale data from being served. It does this by adding a dynamic cache-busting parameter to Catalog Service requests.

When your configuration headers change, the browser fetches fresh data instead of using cached responses. This helps keep your storefront content up to date.
```

> The cache-busting mechanism is handled internally by the boilerplate code:

Reviewer: ```suggestion
Here’s how the boilerplate handles it:
```

> 1. The headers are collected from your [storefront configuration](/setup/configuration/commerce-configuration/) i.e. `public.headers.cs.*`

Reviewer 2: I think technically it collects `headers.all` too, with `getHeaders`.

> 1. The headers are collected from your [storefront configuration](/setup/configuration/commerce-configuration/) i.e. `public.headers.all.*` and `public.headers.cs.*`

Reviewer: ```suggestion
1. It collects headers from your [storefront configuration](/setup/configuration/commerce-configuration/)—`public.headers.all.*` and `public.headers.cs.*`
```

> 2. A short hash (5 characters by default) is generated from these headers

Reviewer: ```suggestion
1. It creates a short hash (5 characters by default) based on those headers.
```

> 3. The hash is automatically appended to the Catalog Service endpoint URL as `?cache-buster=<hash>`

Reviewer 2: If you keep as `cache-buster` this fine, but if you change in PR just remember to change here.

Author: Done.

> 3. The hash is automatically appended to the Catalog Service endpoint URL as `?cb=<hash>`

Reviewer: ```suggestion
1. It adds the hash to the Catalog Service URL using the format `?cb=<hash>`.
```

> 4. The hash updates automatically whenever the headers change

Reviewer: ```suggestion
1. When the headers change, it generates a new hash automatically.
```

> This ensures that any changes to your Catalog Service configuration (like store code, store view code, or other headers) will automatically trigger fresh data fetches.

Reviewer 2: We are doing this specifically because of the CDN configuration, not Catalog Service specifically.

More generally, we should add cache bust to any GET query that we know will be cached by CDN, such as catalog service.

> This ensures that any changes to headers (like store code or store view code) will trigger a fresh data fetch—essential for avoiding stale responses from CDN-cached GET queries such as those hitting the Catalog Service.

Reviewer: ```suggestion
This ensures that any change—like a new store code or view code—triggers a fresh fetch and bypasses CDN-cached GET requests to the Catalog Service.
```

> You don't need to implement anything special to use this feature. The cache-busting is automatically applied when:

Reviewer: ```suggestion
You don’t need to enable or configure anything. The Commerce boilerplate handles everything automatically when:
```

> - The PDP drop-in component initializes

Reviewer: ```suggestion
- The PDP drop-in component initializes.
```

> - Any component makes requests to the Catalog Service endpoint

Reviewer: ```suggestion
- Any component sends a request to the Catalog Service.
```

> - The configuration headers change

Reviewer: ```suggestion
- Your configuration headers update.
```

> The system will automatically handle generating and updating the `cb` parameter as needed.

Reviewer: ```suggestion
The boilerplate always manages the `cb` parameter for you.
```
